src/main/java/com/kodilla/ecommercee/service/UserService.java

```diff
@@ -0,0 +1,63 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.exception.UserNotFoundException;
+import com.kodilla.ecommercee.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Random;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public List<User> getUsers() {
+        return userRepository.findAll();
+    }
+
+    public User getUser(Long id) throws UserNotFoundException {
+        return userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+    }
+
+    public void deleteUser(Long id) throws UserNotFoundException {
+        if(userRepository.findById(id).isPresent()) {
+            userRepository.deleteById(id);
+        } else {
+            throw new UserNotFoundException();
+        }
+    }
+
+    public void updateUser(Long id, User user) throws UserNotFoundException {
+        if(userRepository.findById(id).isPresent()) {
+            userRepository.deleteById(id);
+            userRepository.save(user);
+        } else {
+            throw new UserNotFoundException();
+        }
+    }
+
+    public void switchBlockade(Long id) throws UserNotFoundException {
+        User user = getUser(id);
+        user.switchBlockade();
+        userRepository.save(user);
+    }
+
+    public Integer generateKey(Long id) throws UserNotFoundException {
```

Comment:
also i think we need a method that will be used in requests to check if token is expired

Reply:
The question is whether we need method here or the ability to call getUser is enough for us.
When we verifying user-order, it is not enough for us to validate the token, but also the token number. So we need to download two pieces of information from user object.
Now - in this case isn't it better to download the user object and check on him what the current token number is and is it still valid? Otherwise, we would have to create additional DTOs to transfer tokens with time data and I don't know if it's not overkill...